atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraAbsentHostTracker.java

```diff
@@ -16,22 +16,66 @@
 
 package com.palantir.atlasdb.keyvalue.cassandra;
 
+import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableSet;
 import com.palantir.atlasdb.keyvalue.cassandra.pool.CassandraServer;
+import com.palantir.common.concurrent.PTExecutors;
 import com.palantir.common.streams.KeyedStream;
 import com.palantir.logsafe.SafeArg;
 import com.palantir.logsafe.logger.SafeLogger;
 import com.palantir.logsafe.logger.SafeLoggerFactory;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
 import javax.annotation.concurrent.GuardedBy;
+import one.util.streamex.EntryStream;
 import org.immutables.value.Value;
 
 public final class CassandraAbsentHostTracker {
     private static final SafeLogger log = SafeLoggerFactory.get(CassandraAbsentHostTracker.class);
+    private static final AtomicLong ELEMENTS_MADE_ABSENT = new AtomicLong(0);
+    private static final AtomicLong REMOVED_ELEMENTS = new AtomicLong(0);
+    private static final Set<CassandraAbsentHostTracker> KNOWN_TRACKERS = ConcurrentHashMap.newKeySet();
+    private static final ScheduledExecutorService SCHEDULED_EXECUTOR_SERVICE =
```

Comment:
can this be an instance field and injected in constructor (ideally a WC managed executor)?

Reply:
I was using (abusing?) statics here because I wanted to ensure just one of these runs even on the internal services that have multiple atlasdb clients 😬 This shouldn't merge in this form, in any cawes